src/api/mod.rs

```diff
@@ -80,19 +80,17 @@ impl<T: Transport> Web3Main<T> {
   }
 
   /// Should be used to wait for confirmations
-  pub fn wait_for_confirmations<'a, F, V>(&'a self, poll_interval: Duration, confirmations: usize, check: V)
-  -> Box<Future<Item = (), Error = Error> + 'a> where
-  F: 'a + IntoFuture<Item = Option<U256>, Error = Error>,
-  V: 'a + Fn() -> F,
+  pub fn wait_for_confirmations<F, V>(&self, poll_interval: Duration, confirmations: u64, check: V)
+    -> confirm::Confirmations<&T, V, F::Future> where
```

Comment:
Style nitpick: I would rather see the output type on the same line as the function name and type parameters idented.
If it doesn't fit then it should be split by parameters:
```rust
pub fn wait_for_confirmations<F, V>(
  &self,
  poll_interval: Duration,
  confirmations: u64,
  check: V,
) -> confirm::Confirmations<&T, V, F::Future> where
  F: IntoFuture<..>,
  V: confirm::ConfirmationCheck<Check = F>,
{
 ...
}
```